### PR TITLE
Enable bt_navigator system test on lifecycle branch

### PR DESCRIPTION
**Pull request by mkhansen-intel**
---

## Description of contribution in a few bullet points
 - Enables the bt_navigator system test on the lifecycle branch
    - UPDATE: Now uses the `navigate_w_replanning_and_recovery.xml` behavior tree
 - DISABLES the linters for some packages temporarily

## Future work
 - Enable other behavior trees and change the test to use the default  (DONE)
 - Enable the localization test (#795)
 - Enable the Planner test (was already disabled prior to this PR)
 - Enable the linters again - #812 
 - Clean up the nav2_params.yaml - #813

### CODE REVIEW

File: nav2_system_tests/src/system/CMakeLists.txt [hunk 1]
```diff
@@ -11,6 +11,6 @@ ament_add_test(test_bt_navigator
     TEST_WORLD=${PROJECT_SOURCE_DIR}/worlds/turtlebot3_ros2_demo.world
     TEST_PARAMS=${PROJECT_SOURCE_DIR}/params/nav2_params.yaml
     GAZEBO_MODEL_PATH=${PROJECT_SOURCE_DIR}/models
-    BT_NAVIGATOR_XML=navigate_w_recovery_retry.xml
+    BT_NAVIGATOR_XML=navigate_w_replanning_and_recovery.xml
```
Comment (mhpanah) [inline nav2_system_tests/src/system/CMakeLists.txt]: Our default tree would be `navigate_w_replanning.xml`

Comment (mkhansen-intel) [reply nav2_system_tests/src/system/CMakeLists.txt]: Why no recovery? We have been using it on master branch

File: nav2_system_tests/CMakeLists.txt [hunk 2]
```diff
@@ -41,7 +41,7 @@ if(BUILD_TESTING)
   find_package(ament_cmake_pytest REQUIRED)
 
   # add_subdirectory(src/planning)
-  add_subdirectory(src/localization)
+  # add_subdirectory(src/localization)
```
[the remark this answers is not in the document]
Comment (mkhansen-intel) [reply nav2_system_tests/CMakeLists.txt]: Please make sure it is re-enabled as part of that PR.

**Comment by mkhansen-intel** [comment]
rebased and force pushed - trying to get all latest changes from lifecycle so it will pass CI

**Comment by mkhansen-intel** [comment]
The test passed on one of 3 builds. I think we still have some reliability issues. When it fails in Travis it's due to the map_server service being unavailable for some reason. I suspect that's a FastRTPS issue. I'll retrigger the build and see if it passes.

**Comment by crdelsey** [comment]
@mkhansen-intel is still investigating the travis failure. Do not integrate yet.

**Comment by mkhansen-intel** [comment]
I'm doing a rebase and merge in case we need to revert any commits.

**Pull request merged**